simulation/test_torrent_status.cpp

```diff
@@ -76,35 +99,82 @@ TORRENT_TEST(status_timers)
 			// to stay within that range
 			if (ticks > 20 * 60 * 60)
 			{
-				// unfortunantly we can't test this many ticks
-				TEST_ERROR("This line should be reached but doesn't");
 				return true;
 			}
 
-			if (ticks == 0){
-				// torrent get ready for seeding on first tick, which means +1s
-				startseed_time = lt::clock_type::now();
+			if(tick_is_in_active_range){
+				// 1 second per tick
+				active_time++;
 			}
 
+			// resume on a tick which is far away from next tick check intervall
+			// because with very low upload limit it takes some time to unchock
+			switch(ticks)
+			{
+				case 0:
+					// torrent get ready for seeding on first tick, means time +1s
+					startseed_time = lt::clock_type::now();
+					tick_is_in_active_range = true;
+					break;
+				case 21:
+					handle.pause();
+					tick_is_in_active_range = false;
+					last_active_time = lt::clock_type::now();
+					break;
+				case 64001:
+					handle.resume();
+					tick_is_in_active_range = true;
+					break;
+				case 64061:
+					handle.pause();
+					tick_is_in_active_range = false;
+					last_active_time = lt::clock_type::now();
+					break;
+				case 70001:
+					handle.resume();
+					tick_is_in_active_range = true;
+					break;
+				case 70061:
+					// wait for 72000 or upload finish and check only sometimes
+					ticks_check_intervall = 1000;
+					break;
+			}
 			// verify that the timers seem correct
-			if ((ticks % 10) == 0)
+			if (tick_is_in_active_range && (ticks % ticks_check_intervall) == 0)
 			{
 				lt::time_point now = lt::clock_type::now();
-				auto const since_start = duration_cast<seconds>(now - startseed_time);
 				torrent_status st = handle.status();
-				TEST_CHECK(st.active_duration == since_start);
-				TEST_CHECK(st.seeding_duration == since_start);
-				TEST_CHECK(st.finished_duration == since_start);
+				TEST_CHECK(st.active_duration == active_time);
+				TEST_CHECK(st.seeding_duration == active_time);
+				// TODO finished_duration seems to be broken after time shift
+				TEST_CHECK(st.finished_duration == active_time);
+				printf("active_time: %ld \n", active_time.count());
```

Comment:
instead of printing these values, you could use ``TEST_EQUAL(st.active_duration, active_time);``